posthog/middleware.py

```diff
@@ -360,3 +361,22 @@ def __call__(self, request: HttpRequest):
 
         response = self.get_response(request)
         return response
+
+
+class ExternalProxyMiddleware:
+    """
+    When people run a proxy against PostHog Cloud, they shouldn't be able to use that proxy to login for safety reasons.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request: HttpRequest):
+
+        response: HttpResponse = self.get_response(request)
+        if request.path.strip("/") in ALWAYS_ALLOWED_ENDPOINTS:
+            # In practice CaptureMiddleware has likely already circumvented this middleware, but just incase
+            return response
+        if settings.MULTI_TENANCY and not request.get_host().endswith(".posthog.com"):
```

Comment:
This will not work e.g. on Cloud EU currently, as we require that users add the `Host` header so we'll always get `endswith` returning true. The plan is to also roll that change out to `app.posthog.com`. I'm not 100% sure on the header functionality with proxies but we might need to check [`X-Forwarded-Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host) explicitly.

I'd also expect to be redirected to `https://eu.posthog.com/` in Cloud EU.

Would expect this to be ok with most of current Cloud US though so we could get this in as is but add something to the TODO list to make this work for others.

Reply:
(the alternative could be to update the instructions for adding the redirect to their proxies but I'm not sure how this could work e.g. for CloudFlare CNAME + Proxy)